homeassistant/components/huawei_lte/__init__.py

```diff
@@ -365,9 +365,8 @@ def get_connection() -> Connection:
             ):
                 if not entity_entry.unique_id.startswith("None-"):
                     continue
-                new_unique_id = (
-                    f"{serial_number}-{entity_entry.unique_id.split('-', 1)[1]}"
-                )
+                new_unique_id = entity_entry.unique_id.removeprefix("None-")
+                new_unique_id = f"{serial_number}-{new_unique_id}"
```

Comment:
I guess there is a slight change a behavior here if there is more than one `-`

Reply:
I don't think there is in this case. Note that we previously had `1` as arg for `.split`; we split on the first dash, and keep what's on the right of it (no matter if there are more dashes in there).